TestDerivs/src/test.cxx

```diff
@@ -1,23 +1,89 @@
 #include <loop_device.hxx>
 
 #include <defs.hxx>
+#include <derivs.hxx>
 #include <mat.hxx>
 #include <simd.hxx>
 #include <vec.hxx>
-#include <derivs.hxx>
 
 #include <cctk.h>
 #include <cctk_Arguments.h>
 #include <cctk_Parameters.h>
 
 namespace TestDerivs {
 using namespace Loop;
-using namespace std;
 
 template <typename T, typename VT>
-constexpr void poly(const T kxx, const T kxy, const T kyz, const VT x,
-                    const T y, const T z, VT &u) {
-  u = kxx * x * x + kxy * x * y + kyz * y * z;
+constexpr VT poly(const T kxx, const T kxy, const T kyz, const VT x, const T y,
+                  const T z) {
+  return kxx * x * x + kxy * x * y + kyz * y * z;
+}
+
+template <typename T>
+constexpr void poly_derivs(const T kxx, const T kxy, const T kyz, const T x,
+                           const T y, const T z, Arith::vec<T, dim> &du,
+                           Arith::smat<T, dim> &ddu) {
+  const T sinx = std::sin(x);
+  const T siny = std::sin(y);
+  const T sinz = std::sin(z);
+  const T cosx = std::cos(x);
+  const T cosy = std::cos(y);
+  const T cosz = std::cos(z);
+  du(0) = -2 * kxx * cosx * sinx - kxy * sinx * siny;
+  du(1) = kxy * cosx * cosy + kyz * cosy * sinz;
+  du(2) = kyz * cosz * siny;
+  ddu(0, 0) =
+      -2 * kxx * cosx * cosx + 2 * kxx * sinx * sinx - kxy * cosx * siny;
+  ddu(0, 1) = -kxy * cosy * sinx;
+  ddu(0, 2) = 0.0;
+  ddu(1, 1) = -kxy * cosx * siny - kyz * siny * sinz;
+  ddu(1, 2) = kyz * cosy * cosz;
+  ddu(2, 2) = -kyz * siny * sinz;
+}
+
+template <typename T>
+constexpr Arith::vec<T, dim> beta_profile(const T x, const T y, const T z) {
+  const T vx = 1 * std::sin(x);
+  const T vy = 2 * std::sin(y);
+  const T vz = 3 * std::sin(z);
+  const T norm = std::sqrt(vx * vx + vy * vy + vz * vz);
+  return Arith::vec<T, dim>({vx / norm, vy / norm, vy / norm});
+}
+
+template <typename T>
+constexpr T poly_diss(const T kxx, const T kxy, const T kyz, const T x,
+                      const T y, const T z, const int diss_order,
+                      const Arith::vect<T, dim> &dx) {
+  const T sinx = std::sin(x);
+  const T siny = std::sin(y);
+  const T sinz = std::sin(z);
+  const T cosx = std::cos(x);
+  int coeff = 0;
+
+  switch (diss_order) {
+  case 4:
+    coeff = 8;
+    break;
+  case 6:
+    coeff = -32;
+    break;
+  case 8:
+    coeff = 128;
+    break;
+  case 10:
+    coeff = -512;
+    break;
+  default:
+    assert(0);
+  }
+
+  const int abc = std::abs(coeff);
```

Comment:
ditto

Reply:
in fact you may want to use `fabs` to guard against bad include files which only give you `int abs(int)` and not the overloaded ones as well.